examples/general/non_local_effect_before_error_return/ui/main.rs

```diff
@@ -276,3 +276,15 @@ pub mod public_only {
         Err(VarError::NotPresent)
     }
 }
+
+// Test for the fix in issue #929: functions returning std::fmt::Result should not trigger the lint
+pub mod fmt_result_test {
+    use std::fmt::{self, Write};
+
+    pub fn fmt_result_with_write_before_err_return(buffer: &mut String) -> fmt::Result {
+        // This non-local effect (write) should not trigger a lint warning
+        // because the return type is std::fmt::Result
+        write!(buffer, "Hello, world!")?;
```

Comment:
```suggestion
        buffer.write_str("Hello, world!")?;
```
I tried disabling the fix in `is_disallowed_result` to verify that the lint was firing on this line, and it was not.

After some debugging, I realized that the lint was getting to the `span_lint_and_then` call in lib.rs, the warning just wasn't being shown.

I suspect the warning was not being shown because `fn_span` was in the result of a macro expansion, though I have not confirmed this.

Regardless, changing the call to `write_str` causes the warning to be shown then the fix in `is_disallowed_result` is disabled.